File: leet_easy_java/leet_121/leet_121b.java
class Solution {
    public int maxProfit(int[] prices) {
        int len = prices.length, _min = Integer.MAX_VALUE, profit = 0;
        for(int i=0; i< len; i++){
            _min = Math.min(_min, prices[i]);
            profit = Math.max(profit, prices[i] - _min);
        }
        return profit;
    }
}

File: leet_easy_javascript/leet_121/leet_121b.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var profit = 0, val = prices[0];
    for(var i=1; i<prices.length; i++){
        profit = Math.max(profit, prices[i] - val);
        val = Math.min(val, prices[i]);           
    }
    return profit;
};